osu.Game/Rulesets/Edit/Checks/CheckSongFormat.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using ManagedBass;
+using osu.Framework.Audio.Callbacks;
+using osu.Game.Beatmaps;
+using osu.Game.Extensions;
+using osu.Game.Rulesets.Edit.Checks.Components;
+
+namespace osu.Game.Rulesets.Edit.Checks
+{
+    public class CheckSongFormat : ICheck
+    {
+        public CheckMetadata Metadata => new CheckMetadata(CheckCategory.Audio, "Checks for song formats.");
+
+        public IEnumerable<IssueTemplate> PossibleTemplates => new IssueTemplate[]
+        {
+            new IssueTemplateFormatUnsupported(this),
+            new IssueTemplateIncorrectFormat(this),
+        };
+
+        private IEnumerable<ChannelType> allowedFormats => new ChannelType[]
+        {
+            ChannelType.MP3,
+            ChannelType.OGG,
+        };
+
+        public IEnumerable<Issue> Run(BeatmapVerifierContext context)
+        {
+            var beatmapSet = context.Beatmap.BeatmapInfo.BeatmapSet;
+            var audioFile = beatmapSet?.GetFile(context.Beatmap.Metadata.AudioFile);
+
+            if (beatmapSet == null) yield break;
+            if (audioFile == null) yield break;
+
+            using (Stream data = context.WorkingBeatmap.GetStream(audioFile.File.GetStoragePath()))
+            {
+                if (data == null || data.Length <= 0) yield break;
+
+                var fileCallbacks = new FileCallbacks(new DataStreamFileProcedures(data));
+                int decodeStream = Bass.CreateStream(StreamSystem.NoBuffer, BassFlags.Decode | BassFlags.Prescan, fileCallbacks.Callbacks, fileCallbacks.Handle);
```

Comment:
Prescan can add considerable overhead, and doesn't look like it would be required in this usage.

Reply:
Also I think you'll want to free the bass stream after use (`Bass.StreamFree()`).

Reply:
done